common/lib/Warewulf/File.pm

```diff
@@ -427,6 +428,15 @@ sync()
             $self->checksum($digest->hexdigest());
             $self->size($total_len);
             $db->persist($self);
+            if ( $metadata ) {
```

Comment:
Can you move the  `$db->persist($self);` call _below_ the entire `if ( $metadata ) { ... }` block?

`$db->persist()` is what stores the information back into the database, otherwise you're just changing the information on the working object.

Reply:
Good catch.  I will do that.